presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1220,6 +1233,13 @@ protected Scope visitSetOperation(SetOperation node, Optional<Scope> scope)
             return createAndAssignScope(node, scope, outputDescriptorFields);
         }
 
+        private boolean isExpensiveUnionDistinct(SetOperation setOperation, Type[] outputTypes)
+        {
+            return setOperation instanceof Union && setOperation.isDistinct() && outputTypes.length > UNION_DISTINCT_FIELDS_WARNING_THRESHOLD
```

Comment:
Since we are updating the change, let's also reformat this:
* The codebase normally put `&&` or `||` at the end of the previous line rather than beginning of next.
* streaming API would put each method on a new line after `.stream()`
* Personally if the predicates don't all fit in one line I always break them to be one line each

So something like the following:
```
    return setOperation instanceof Union &&
            setOperation.isDistinct() &&
            outputTypes.length > UNION_DISTINCT_FIELDS_WARNING_THRESHOLD &&
            Arrays.stream(outputTypes)
                    .anyMatch(
                            type -> type instanceof RealType ||
                                    type instanceof DoubleType ||
                                    type instanceof MapType ||
                                    type instanceof ArrayType ||
                                    type instanceof RowType);
```